var/spack/repos/builtin/packages/py-heat/package.py

```diff
@@ -34,6 +34,11 @@ class PyHeat(PythonPackage):
         default=False,
         description="Use py-scikit-learn and py-matplotlib for the example tests",
     )
+    variant("cuda", default=False, description="Build with CUDA support")
+    variant("cuda_arch", default="80", description="CUDA compute capability")
```

Comment:
When overridden like this, `cuda_arch` is wrong, since it doesn't allow any value. What was the issue here, for which we can't inherit these variants from `CudaPackage`?

Reply:
thanks @alalazo . 

The initial version of this recipe (`class PyHeat(PythonPackage):`...) was building py-heat with py-torch-cpu by default. We would like to be able to build py-heat with the cuda or rocm version of py-torch, depending on the system.

I'm not sure what the best way of doing it is, I'm grateful for any help. Thanks!